# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/ron/tests/407_raw_value.rs
# Commit: 3f167f8b3137
# Full Hash: 3f167f8b313749f50f135e189e5854bef7ac0c73
# Author: Erich Gubler <erichdongubler@gmail.com>
# Date: 2025-10-03 18:08 +0000
# Regressor Bug: 1991551
# File Overlap Count: 23
# Overlapping Files: third_party/rust/wgpu-hal/Cargo.toml, supply-chain/audits.toml, third_party/rust/naga/src/back/spv/writer.rs, third_party/rust/naga/src/back/msl/writer.rs, third_party/rust/wgpu-types/src/lib.rs
# Description:
#   Bug 1991551 - build(webgpu): update WGPU to 8c4aebc0c1b75e3f7412dfc180745a0717fc30ec r=webgpu-reviewers,supply-chain-reviewers,nical
# ==============================================================================

diff -r 823439f307a0 -r 3f167f8b3137 third_party/rust/ron/tests/407_raw_value.rs
--- a/third_party/rust/ron/tests/407_raw_value.rs	Fri Oct 03 18:01:30 2025 +0000
+++ b/third_party/rust/ron/tests/407_raw_value.rs	Fri Oct 03 18:08:29 2025 +0000
@@ -1,6 +1,6 @@
 use ron::{
     de::from_bytes,
-    error::{Error, Position, SpannedError},
+    error::{Error, Position, Span, SpannedError},
     from_str, to_string,
     value::RawValue,
 };
@@ -47,7 +47,10 @@
         err,
         SpannedError {
             code: Error::TrailingCharacters,
-            position: Position { line: 1, col: 3 }
+            span: Span {
+                start: Position { line: 1, col: 3 },
+                end: Position { line: 1, col: 3 }
+            }
         }
     );
 
@@ -56,7 +59,10 @@
         err,
         SpannedError {
             code: Error::UnexpectedChar('\0'),
-            position: Position { line: 1, col: 1 }
+            span: Span {
+                start: Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 1 }
+            }
         }
     )
 }
@@ -71,7 +77,10 @@
         err,
         SpannedError {
             code: Error::TrailingCharacters,
-            position: Position { line: 1, col: 3 }
+            span: Span {
+                start: Position { line: 1, col: 3 },
+                end: Position { line: 1, col: 3 }
+            }
         }
     );
 
@@ -85,7 +94,10 @@
         err,
         SpannedError {
             code: Error::Eof,
-            position: Position { line: 1, col: 2 },
+            span: Span {
+                start: Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 2 },
+            }
         }
     );
 }
@@ -108,7 +120,10 @@
         err,
         SpannedError {
             code: Error::ExpectedInteger,
-            position: Position { line: 1, col: 10 },
+            span: Span {
+                start: Position { line: 1, col: 9 },
+                end: Position { line: 1, col: 10 },
+            }
         }
     );
 }
@@ -214,7 +229,7 @@
     assert_eq!(
         err,
         Error::Message(String::from(
-            "invalid RON value at 1:2: Unexpected end of RON"
+            "invalid RON value at 1:1-1:2: Unexpected end of RON"
         ))
     );
 }
@@ -273,7 +288,10 @@
         RawValue::from_ron(""),
         Err(SpannedError {
             code: Error::Eof,
-            position: Position { line: 1, col: 1 },
+            span: Span {
+                start: Position { line: 1, col: 1 },
+                end: Position { line: 1, col: 1 },
+            }
         })
     );
 
@@ -287,7 +305,10 @@
             code: Error::Message(String::from(
                 "ron::value::RawValue cannot enable extensions"
             )),
-            position: Position { line: 1, col: 27 },
+            span: Span {
+                start: Position { line: 1, col: 27 },
+                end: Position { line: 1, col: 27 },
+            }
         })
     );
 
@@ -297,7 +318,10 @@
             code: Error::Message(String::from(
                 "ron::value::RawValue cannot enable extensions"
             )),
-            position: Position { line: 1, col: 27 },
+            span: Span {
+                start: Position { line: 1, col: 27 },
+                end: Position { line: 1, col: 27 },
+            }
         })
     );
 
@@ -305,14 +329,20 @@
         RawValue::from_ron("42 //"),
         Err(SpannedError {
             code: Error::UnclosedLineComment,
-            position: Position { line: 1, col: 6 },
+            span: Span {
+                start: Position { line: 1, col: 6 },
+                end: Position { line: 1, col: 6 },
+            }
         })
     );
     assert_eq!(
         ron::from_str::<&RawValue>("42 //"),
         Err(SpannedError {
             code: Error::UnclosedLineComment,
-            position: Position { line: 1, col: 6 },
+            span: Span {
+                start: Position { line: 1, col: 6 },
+                end: Position { line: 1, col: 6 },
+            }
         })
     );
     assert_eq!(
@@ -332,14 +362,20 @@
         RawValue::from_ron("a//"),
         Err(SpannedError {
             code: Error::UnclosedLineComment,
-            position: Position { line: 1, col: 4 },
+            span: Span {
+                start: Position { line: 1, col: 4 },
+                end: Position { line: 1, col: 4 },
+            }
         })
     );
     assert_eq!(
         ron::from_str::<&RawValue>("a//"),
         Err(SpannedError {
             code: Error::UnclosedLineComment,
-            position: Position { line: 1, col: 4 },
+            span: Span {
+                start: Position { line: 1, col: 4 },
+                end: Position { line: 1, col: 4 },
+            }
         })
     );
     assert_eq!(